scripts/check_ecosystem.py

```diff
@@ -45,11 +45,11 @@ async def clone(self: Self, checkout_dir: Path) -> AsyncIterator[Path]:
         """Shallow clone this repository to a temporary directory."""
         if checkout_dir.exists():
             logger.debug(f"Reusing {self.org}:{self.repo}")
-            yield Path(checkout_dir)
+            yield await self._get_commit(checkout_dir)
```

Comment:
Can we add the commit SHA to the `Repository` object as that's where I think it should be logically? I think adding it to `ref` should be the same meaning as branches are just pointers to specific commits.

Reply:
This will also mean that the `repo.url_for` function/call need not be changed.

Reply:
I tried that first but we cannot because it's immutable. We definitely could decide to make it _mutable_, but I liked the immutable design. More commentary at https://github.com/astral-sh/ruff/pull/5704#discussion_r1260592025

Reply:
Oh, I missed that comment. Give me a minute.